### ISSUE TITLE
deleting a domain fails

### ISSUE THREAD

**Issue by thomas-mc-work** [payara/docker-payaraserver-full#8]
I'm trying to delete the existing domains which fails. I've inserted this in line 17:

    RUN $PAYARA_PATH/bin/asadmin delete-domain domain1

The error message is

> A file or folder within domain domain1 at /opt/payara41/glassfish/domains/domain1 is in use. Stop using it before deleting the domain.

The same happens when deleting the payaradomain. When i download and extract the same archive on my local machine (debian based Xubuntu 16.04) then the deletion works fine.

**Comment by thomas-mc-work** [comment]
ping @mikecroft

**Comment by mikecroft** [comment]
Hi @thomas-mc-work sorry I missed this - is domain1 running at the point you try to delete it? I'm not sure exactly what could be happening here, it could be a timing thing. Can you link to the full Dockerfile you're using so I can see for myself?

**Comment by thomas-mc-work** [comment]
Here you go:

    FROM openjdk:8-jdk

    RUN \
     apt-get update && \ 
     apt-get install -y unzip 

    ENV ADMIN_USER admin

    ENV PAYARA_PATH /opt/payara41

    RUN \ 
     mkdir -p ${PAYARA_PATH}/deployments && \
     useradd -b /opt -m -s /bin/bash -d ${PAYARA_PATH} payara && echo payara:payara | chpasswd

    # specify Payara version to download
    ENV PAYARA_PKG https://s3-eu-west-1.amazonaws.com/payara.fish/Payara+Downloads/Payara+4.1.2.172/payara-4.1.2.172.zip
    ENV PAYARA_VERSION 172

    ENV PKG_FILE_NAME payara-full-${PAYARA_VERSION}.zip

    # Download Payara Server and install
    RUN \
     wget --quiet -O /opt/${PKG_FILE_NAME} ${PAYARA_PKG} && \
     unzip -qq /opt/${PKG_FILE_NAME} -d /opt && \
     chown -R payara:payara /opt && \
     # cleanup
     rm /opt/${PKG_FILE_NAME}

    USER payara
    WORKDIR ${PAYARA_PATH}

    # set credentials to admin/admin 

    ENV ADMIN_PASSWORD admin

    RUN echo 'AS_ADMIN_PASSWORD=\n\
    AS_ADMIN_NEWPASSWORD='${ADMIN_PASSWORD}'\n\
    EOF\n'\
    >> /opt/tmpfile

    RUN echo 'AS_ADMIN_PASSWORD='${ADMIN_PASSWORD}'\n\
    EOF\n'\
    >> /opt/pwdfile

    RUN ${PAYARA_PATH}/bin/asadmin delete-domain domain1

**Comment by simdevmon** [comment]
I am having the same issue using `Payara 4.1.2.172` with `Docker version 17.05.0-ce, build 89658be`.
domain1 or payaradomain are not running at that point.

**Comment by mikecroft** [comment]
Sorry for the delay, I've confirmed this is the case. The `delete-domain` command simply deletes the domain directory, so a valid workaround is to replace that command with `RUN rm -rf ${PAYARA_PATH}/glassfish/domains/domain1 `

The command is [failing on a check that the domain directory can be renamed](https://github.com/payara/Payara/blob/master/nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/DeleteDomainCommand.java#L131-L132) (the message string is quite misleading here) so that would suggest a permissions issue. I tried setting permissions to 777 before the delete with no success. Since this message gets printed out if *anything* fails, this could even be a problem with any part of that method. I'll keep investigating and see what I can find.
